Merge to master to make 14.2.0 release

ehoogerbeets opened the pull request:
### Checklist

* [x] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [ ] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [ ] This PR has passed all of the test cases on `QT/QML`.
* [x] This is an API breaking change.
* [ ] Requires a major version change.


### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

ehoogerbeets committed:
Now depends on cldr 34

ehoogerbeets committed:
Make genclockprefs.js use the json in cldr-data

ehoogerbeets committed:
Update to use cldr-data where possible

ehoogerbeets committed:
Update to use cldr-data where possible

ehoogerbeets committed:
Use cldr-data where possible

ehoogerbeets committed:
Use cldr-data where possible

ehoogerbeets committed:
Update gendatefmts2 and update copyright notices

ehoogerbeets committed:
Update gendatefmts2 and update copyright notices

ehoogerbeets committed:
Converted more tools to use cldr-data

ehoogerbeets committed:
Converted more tools to use cldr-data

ehoogerbeets committed:
Updated to latest versions of various packages.

ehoogerbeets committed:
Remove the DOS EOF characters

ehoogerbeets committed:
Converted genweekdata

ehoogerbeets committed:
Converted genpapersizes.js to use cldr-data

ehoogerbeets committed:
Updated gennumfmt.js to use cldr-data

gouniLee committed:
Updated the Taiwan area code of PhoneNumber according to Wikipedia

gouniLee committed:
Fix 3 digit iddprefix format of zh-Hant-TW

gouniLee committed:
Change version to prepare for the next release

gouniLee committed:
Fix main file in package.json

ehoogerbeets committed:
Merge pull request #166 from iLib-js/twPhonefmt

Update the Taiwan area code of Phone Number according to Wikipedia

ehoogerbeets committed:
Add tempDir env var

Try to fix the bogus output dir for webpack

ehoogerbeets committed:
Fix the output dir for the locale temp files

This used to go in js/ilib, but now it goes into
js/output/locales so that it gets cleaned up with
ant clean.

ehoogerbeets committed:
Depend on the latest ilib webpack loader and plugin

ehoogerbeets committed:
Now uses the independent ilib-scanner

Instead of the one that is in ilib-webpack-loader, which is going
away.

ehoogerbeets committed:
Last thing that needs to be done.

Now "allscripts.sh" will update to the latest
version of CLDR, whatever that may be.

Still need to do something similar for the
Unicode character database.

ehoogerbeets committed:
Put this back the way it was in the development branch.

ehoogerbeets committed:
Fixed a few of the scripts

Didn't work as is or didn't put the output in the right
places.

ehoogerbeets committed:
A few more fixes to everything running correctly

ehoogerbeets committed:
Update all data to CLDR 34 and UCD 12

ehoogerbeets committed:
Fix many unit tests to correspond to CLDR 34/UCD 12

ehoogerbeets committed:
Fixed the strings-ext unit tests

ehoogerbeets committed:
Revert the region name updates

- CLDR for some unfathomable reason changed the region
codes for various regions. This is not acceptable, as
it means that databases that depend on stable codes have
bogus data now.
- Various territory regions were missing from the US and
other countries.

Not sure what kind of crap they were doing!

ehoogerbeets committed:
Work around problem in cldr-data

The json was generated with an extra space in the currency format
for fr-CH. It seems to be a conversion error between the CLDR
XML, which is canonical, and the json in cldr-data, which is
derivative. The XML did not have the space, and the json data
introduced it.

The solution was to trim the white space from the
start and end of all number formats to work around that problem.

ehoogerbeets committed:
Changed version number and added release notes

ehoogerbeets committed:
Merge pull request #169 from iLib-js/updateNpmModules

Updated to latest versions of various packages.

ehoogerbeets committed:
Merge branch 'development' into fixOutputDir

ehoogerbeets committed:
Merge pull request #168 from iLib-js/newScanner

Now uses the independent ilib-scanner

ehoogerbeets committed:
Merge branch 'development' into fixOutputDir

ehoogerbeets committed:
Merge branch 'development' into cldr34

ehoogerbeets committed:
Merge branch 'cldr34' into cldr34-data

ehoogerbeets committed:
Merge pull request #167 from iLib-js/fixOutputDir

Fix the output dir for the locale temp files

ehoogerbeets committed:
Was missing some functions to make assert.deepEqual work properly

ehoogerbeets committed:
Merge pull request #172 from iLib-js/fixAssertExtras

Was missing some functions to make assert.deepEqual work properly

ehoogerbeets committed:
Start updating the readme

ehoogerbeets committed:
Updates to the readme and mentions other ilib projects

ehoogerbeets committed:
Merge pull request #173 from iLib-js/updateReadme

Update readme

ehoogerbeets committed:
Get the right target dir parameter!

ehoogerbeets committed:
Clean up files first before generating new ones

That allows us to delete files that are no longer generated.
Also, ensure that the cldr-data coverage is "full" for all the
locales that CLDR has.

ehoogerbeets committed:
Merge branch 'cldr34' into cldr34-data

gouniLee committed:
Add test files to verify CLDR 34. Some expected value is LG overriding one. I marked them. those should be updated

gouniLee committed:
Fix some test function name and locale. Some of them were miswritten

gouniLee committed:
Add test file to verify date/time format. fix expeced result in fa-AF Monthth translation

ehoogerbeets committed:
Move most of the LG tests from the qa dir to their right place

The tests do not pass yet. The work to make them pass
is yet to be done.

ehoogerbeets committed:
Remove some tests for LG overriding data

The open-source should not have these overrides.

ehoogerbeets committed:
Update currencies based on CLDR 34

gouniLee committed:
Update test expected value to cldr instead LG overriing one.Add more testcase about date/time and number(normal, currency,percentage)

gouniLee committed:
Update copyright. Remove LG

gouniLee committed:
Add testfile to verify daterange format template. Fix some testfunction name ku_KZ-> ru_KZ

gouniLee committed:
Write DateRangeFormat Spec doc. Those were not defined in CLDR

gouniLee committed:
Add testdurfmt2.js to test Duration Format.

Tested numbers are different per locale. Because plural category is different per lanauge.
All of medium length testase is commented out. Those will be update after data is properly created

gouniLee committed:
Update lg testexpected valued to CLDr34

gouniLee committed:
Add more condition to creat currency.js annd update copyright

Add more condition to extract default currency information.
In order to filter _tender option.
CLDR provide '_tender' option. it indicates whether or not the ISO
currency code represents a currency that was or is legal tender
in some country. Certain ISO codes represent things like
financial instrunments or precious metal, and do not represnet normally
interchanged currencies.

ehoogerbeets committed:
Set cldr coverage to "full"

ehoogerbeets committed:
Fix indentation

ehoogerbeets committed:
Merge branch 'cldr34' into cldr34-data

ehoogerbeets committed:
Merge pull request #176 from iLib-js/cldrFull

Set cldr coverage to "full"

ehoogerbeets committed:
Merge branch 'development' into cldr34

ehoogerbeets committed:
Merge branch 'cldr34' into cldr34-data

ehoogerbeets committed:
Merge branch 'development' into cldr34

ehoogerbeets committed:
Merge branch 'cldr34' into cldr34-data

gouniLee committed:
Update und/CN/currency.jf. The latest result of currency tool code

gouniLee committed:
Re-update firstdayofweek.jf, weekendstartjf and weekendend.jf files. genweekdate.js tool code doesn't have problem. The result of running it, it includes CLDR 34 changes. It seems to have a problme while file copy process. tests from testDayOfWeek.js, passed all of them

gouniLee committed:
Re copy all files from the result of genclockprefs.js

Passed all the testcase in testclock.js

gouniLee committed:
Modigy genclockprefs.js code to look at ckb/ when a ku language

gouniLee committed:
Dele all numfmt.js files then copy again from running gennumfmt.js. and regarding ku/numfmt.jf, I manually coped from ckb/numfmt. and Fix some test failures

gouniLee committed:
Fix current number testcase failures based on newly updated CLDR 34 data

gouniLee committed:
Fix curent test failures on root category based on newly update CLDR 34

gouniLee committed:
Fix test failures in units. decimal period makes and error in french. eventhough values are same

gouniLee committed:
Add ilib-web.js for pure Web App support and Fix index.js for webOS Platform

gouniLee committed:
Simplified PR template

gouniLee committed:
Update ReleaseNotes

gouniLee committed:
Delete all dateformat.json & sysres.json then copy all of files again from runing gendatefmt.js

gouniLee committed:
Rollback dateformat.json & sysres.json. I realize getdatefmt2.js tool code doesn't create relative date/time information data. So should not be removed files before copying all files from tool code result.

gouniLee committed:
Make the generated data to be merged correctly

gouniLee committed:
Delete all dateformats.json & sysres.json then copy all of files again from running gendatefmt2.js. Now relative format data created properly

gouniLee committed:
Fix some test failures in date feature

gouniLee committed:
Fix test failures in DateFmt feature

gouniLee committed:
Remove ha in rtl language list. Enable to create duration medium locale data, whcih is same with short.

gouniLee committed:
Re-generate sysres.json. Creating a medium length format of duration from tool code. which weren't

gouniLee committed:
Rollback cop/clock.jf,dateformats.json,sysres.json and no/dateforamts.json,sysres.json. Those language data don't exist in CLDR

gouniLee committed:
More fix testfailures on date & daterange

gouniLee committed:
Fix test failures on datefmtrange feature

gouniLee committed:
Fix tool code bug. If a code check the length of value, '' case values are deleted. The problem found when creating listPattern value. Chinese and Japanese are having '' cases

gouniLee committed:
Update zh/sysres.json, ja/sysres.json. seperator values were removed while generating value through tool code. Manual korean millisecond value

gouniLee committed:
Fix more test failures in duration formating feature

gouniLee committed:
In ms, kk language, I add variation code to put an arabic script in the last order

gouniLee committed:
Update kk/scripts.jf, ms/scripts.js. arabic script would be in the last

gouniLee committed:
Fix test failures in duration fmt in ms,kk language

gouniLee committed:
Add missing locale in build.xml (es-PH, is-IS, tr-CY,de-CH), Add files whech are newly created to the list for the brwoser test

gouniLee committed:
Add 'webos-webapp' platform return type for webos web application

ehoogerbeets committed:
Fix unit tests to use the right calendar for fa-IR & fa-AF

gouniLee committed:
Update ku/sysres.json. Some of the data will keep the current iLib data rather than CLDR. Enable all of test regarding ku-Arab-IQ and fix some cases. In ku-Arab-IQ, default number style is arabic style.

gouniLee committed:
Update ReleaseNotes

gouniLee committed:
Enable all of test cases regarding medium length duration formatting and fixed them all

ehoogerbeets committed:
Fix the map from persian-algo to gregorian

For some reason, the name of the persian-algo calendar
got overwritten with a long dash instead of the
regular ASCII dash. This puts it back again.

ehoogerbeets committed:
Merge branch 'cldr34-data' of github.com:iLib-js/iLib into cldr34-data

gouniLee committed:
Add timezone:'Etc/UTC' for Farsi testcases in weekdayTranslation. Now Every cases 100% passed

gouniLee committed:
Add test case to verify weekday translation when a calendar is persian. and Added some description

ehoogerbeets committed:
Merge pull request #170 from iLib-js/cldr34

CLDR update scripts

ehoogerbeets committed:
Merge branch 'development' into cldr34-data

ehoogerbeets committed:
Merge pull request #171 from iLib-js/cldr34-data

Update to CLDR 34 and UCD 12.0

gouniLee committed:
Merge branch 'development' into supportWebOS

ehoogerbeets committed:
Merge pull request #177 from iLib-js/supportWebOS

Add ilib-web.js to support pure webapp and fix index.js to support webOS properly

gouniLee commented:
Please don't merge this PR. There're some test failures on QT/QTML. (Duration and  Daterange features)

ehoogerbeets committed:
Re-introduce merged data caching for QT

- other platforms can use it as well, but only QT does it
by default

gouniLee committed:
Re-write some testcase in order to work on qt

gouniLee committed:
Add guard code to prevent false setting in qt. When data lodads, I found that value is false on qt

gouniLee committed:
Check ilib._cacheMerged value again in Utils.js.

gouniLee committed:
Update ReleaseNotes

gouniLee committed:
Update testfiles. QT doesn't allow global variable. I add a function to store variable, then testfunction call this function instead

gouniLee committed:
Make ilib._cachedMerged value is set in ilib.js

gouniLee committed:
Update cldr tool code to include data which doesn't exist in CLDR

gouniLee committed:
Fix bug

gouniLee committed:
Fix one more cases

gouniLee committed:
Update ReleaseNotes

gouniLee committed:
Changed a way of setting value to array

gouniLee committed:
Add ilib-web.js to uglify task list and minor update

gouniLee committed:
Update log

ehoogerbeets committed:
Merge pull request #183 from iLib-js/fixSome

Added `ilib-web.js' to uglify task list and minor update

ehoogerbeets committed:
Merge pull request #182 from iLib-js/fixcldrTool

Implement to include automatically data which doesn't exist in CLDR

ehoogerbeets committed:
Fix warnings about "new Buffer is deprecated"

The new way to do it is to call the Buffer.from()
factory method instead.

ehoogerbeets committed:
Make both the regular node and qt test suites work properly.

ehoogerbeets committed:
Merge branch 'development' into cacheTesting

gouniLee committed:
Make all of testfile to run qt. Precious it wasn't.. Re-write test case to run testfiles on qt (month/weekday translation testfile)

gouniLee committed:
Merge branch 'development' into fixTestcaseform

ehoogerbeets committed:
Merge pull request #180 from iLib-js/cacheTesting

Re-introduce merged data caching for QT

ehoogerbeets committed:
Merge branch 'development' into fixWarnings

gouniLee committed:
Change the file order to make pass all the phone test on qt. It's also related global variable issue

gouniLee committed:
Merge branch 'development' into fixTestcaseform

gouniLee committed:
Fix test failures on QT

ehoogerbeets committed:
Merge pull request #181 from iLib-js/fixTestcaseform

Update some test form in order to work properly on qt

ehoogerbeets committed:
Merge pull request #184 from iLib-js/fixWarnings

Fix warnings from "new Buffer"

ehoogerbeets committed:
Try fixing the travis build

ehoogerbeets committed:
Try fixing the travis build

ehoogerbeets committed:
For Travis, only install and test java and js

QT is not installed (which makes the builds faster!) and
not tested.

ehoogerbeets committed:
For Travis, only install and test java and js

QT is not installed (which makes the builds faster!) and
not tested.

ehoogerbeets committed:
v14.1.2 -> v14.2.0

ehoogerbeets committed:
Merge pull request #186 from iLib-js/updateVersion

v14.1.2 -> v14.2.0

ehoogerbeets committed:
Fix unit test failure for version number

ehoogerbeets committed:
Don't do export in the qt subdir

That way, it won't try to build the filereader using qt.
Also, fixed the qt clean so that it doesn't fail if there is no
qt installed.

ehoogerbeets committed:
Merge branch 'fixBuild' of github.com:iLib-js/iLib into fixBuild

ehoogerbeets committed:
Work around a bug in node 10.16.0

The bug is that an undefined or zero-length TZ environment variable
is considered to be an invalid time zone instead of just "no time
zone". Other versions of node do not have this bug. Only 10.16.0
does. Grrr.

ehoogerbeets committed:
Merge pull request #185 from iLib-js/fixBuild

Try fixing the travis build
